satellite/satellitedb/usercredits.go

```diff
@@ -85,7 +87,13 @@ func (c *usercredits) Create(ctx context.Context, userCredit console.UserCredit)
 		referrerID = userCredit.ReferredBy[:]
 	}
 
-	result, err := c.db.DB.ExecContext(ctx, c.db.Rebind(statement), userCredit.UserID[:], userCredit.OfferID, userCredit.CreditsEarned.Cents(), userCredit.ExpiresAt, referrerID, new([]byte), userCredit.OfferID, userCredit.OfferID)
+	var result sql.Result
+	var err error
+	if c.tx != nil {
+		result, err = c.tx.Tx.ExecContext(ctx, c.db.Rebind(statement), userCredit.UserID[:], userCredit.OfferID, userCredit.CreditsEarned.Cents(), userCredit.ExpiresAt, referrerID, new([]byte), userCredit.OfferID, userCredit.OfferID)
+	} else {
+		result, err = c.db.DB.ExecContext(ctx, c.db.Rebind(statement), userCredit.UserID[:], userCredit.OfferID, userCredit.CreditsEarned.Cents(), userCredit.ExpiresAt, referrerID, new([]byte), userCredit.OfferID, userCredit.OfferID)
```

Comment:
Ugh, DBX makes this ugly, doesn't it. There isn't a common-interface way to get the `DB` member from `c.tx` and `c.db`, so you have to have an if-else.

One thing you could do- just to reduce the risk of one of these calls changing and not matching the other- is use a temporary interface type to unify the `ExecContext()` calls, if not the part about getting the DB handle. I think it would look something like

```go
    var dbExec interface {
        ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
    }
    if c.tx != nil {
        dbExec = c.tx.Tx
    } else {
        dbExec = c.db.DB
    }
    result, err = dbExec.ExecContext(ctx, .....)
```

It's not a _lot_ better, though, granted. Only a suggestion.